components/src/dynamo/profiler/utils/dgdr_validate.py

```
@@ -124,53 +124,44 @@ def _validate_sla(sla: SLASpec) -> None:
+                raise ValueError(
+                    "pre_deployment_sweeping_mode in PlannerConfig cannot be 'none' when enable_throughput_scaling is enabled. "
+                    "Throughput-based scaling requires pre-deployment sweeping to generate engine performance data."
+                )
+            elif (
+                planner_sweep_mode == PlannerPreDeploymentSweepMode.Rapid
+                and not aic_supported
+            ):
+                raise ValueError(
+                    f"AIC does not support {dgdr.model} on {dgdr.hardware.gpuSku.lower()} and {dgdr.backend}"
+                    "pre_deployment_sweeping_mode in PlannerConfig can only be 'thorough' when AIC does not support the model/hardware/backend combination. "
+                )
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)